src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java

```diff
@@ -553,4 +553,46 @@ public void setNewGameMode(GameType newGameMode)
             this.newGameMode = newGameMode;
         }
     }
+
+    /**
+     *
+     * Fired when a player clicks the "done" button in the sign edit UI.
+     * Cancelling will set the edited sign to be blank (i.e. lines will be an empty list)
```

Comment:
This gives the impression that this event is fired on the client side, with the reference to the sign editing screen. Please mention in the javadocs which [logical side](https://mcforge.readthedocs.io/en/1.18.x/concepts/sides/) this event is fired on.

Furthermore, looking at the code suggests that this event is fired whenever any player sends a valid sign change packet to the server, even if the change is rejected later in `SeverGamePacketListenerImpl#updateSignText` due to the sign not being editable to that user or the block not even being a sign. 
It bears worth mentioning that aspect in this event, and possibly even adding a helper method to this event for checking if the sign block entity is actually editable by the player in this event (`isEditableByPlayer`). My main concern for this is a mod which naively logs all sign changes: this could possibly allow malicious actors to spam any such logs.

Reply:
Yep I completely agree and I hadn't noticed that I had avoided the later checks. Although now looking at it this could make it more useful as it provides a wider range of use cases. I've added the isEditableByPlayer and commented about the logical side (and updated the description to include it firing after receiving the packet)